docs/how-to/purge-data-on-shadow-table.rst

```diff
@@ -0,0 +1,30 @@
+==========================================
+Purge data on shadow tables
+==========================================
+
+The purge step is disable by default. When enabled, it will run before proceeding to actual upgrade steps.
+This can be enabled with the ``--purge`` flag.
+
+This purge step is a performance optimisation, delete data from the shadow tables in nova database. The behaviour is equal to run juju action ``purge`` on nova-cloud-controller unit, which help to reduce the size of the database, make queries faster, backups efficiency, and follow the data retention policies.
+
+The ``purge-before-date`` flag is supported to delete the data older than the date provided. The date string format should be YYYY-MM-DD[HH:mm][:ss]. This flag is useful to retain recent data for debugging or data retention policies.
+
+
+Usage examples
+--------------
+
+.. terminal::
+   :input cou plan --purge --purge-before-date 2000-01-02
+
+
+More information
+----------------
+
+- `nova-cloud-controller charm actions`_
+- `nova-manage reference`_ - see `purge` subcommand
+- :doc:`Archive old data <how-to/archive-old-data>`
+- :doc:`Nova data migration <explanation/nova-data-migration>`
```

Comment:
Could you check the doc links please?  When I build the docs, sphinx emits some warnings:

```
/home/ubuntu/proj/canonical/charmed-openstack-upgrader/docs/explanation/nova-data-migration.rst:78: WARNING: unknown document: 'how-to/archive-old-data'
/home/ubuntu/proj/canonical/charmed-openstack-upgrader/docs/explanation/nova-data-migration.rst:79: WARNING: unknown document: 'how-to/purge-data-on-shadow-table'
/home/ubuntu/proj/canonical/charmed-openstack-upgrader/docs/how-to/archive-old-data.rst:57: WARNING: unknown document: 'how-to/purge-data-on-shadow-table'
/home/ubuntu/proj/canonical/charmed-openstack-upgrader/docs/how-to/archive-old-data.rst:58: WARNING: unknown document: 'explanation/nova-data-migration'
/home/ubuntu/proj/canonical/charmed-openstack-upgrader/docs/how-to/purge-data-on-shadow-table.rst:25: WARNING: unknown document: 'how-to/archive-old-data'
/home/ubuntu/proj/canonical/charmed-openstack-upgrader/docs/how-to/purge-data-on-shadow-table.rst:26: WARNING: unknown document: 'explanation/nova-data-migration'
```